src/rust/ensogl/src/math/geometry/circle/circle_segment.rs

```diff
@@ -0,0 +1,172 @@
+//! Provides functionality related to circle segments.
+
+use crate::math::algebra::{Cos,Sin,Field};
+
+use core::fmt::Debug;
+
+
+
+// =====================
+// === CircleSegment ===
+// =====================
+
+/// Implements computations related to circle segments. For details and background on the formulas
+/// used, see https://en.wikipedia.org/wiki/Circular_segment
+#[derive(Clone,Debug,PartialEq)]
+pub struct CircleSegment<T> {
+    radius : T,
+    angle  : T,
+}
+
+pub trait FloatLike<T> = Field<T> + Sin<Output=T> + Cos<Output=T> + From<f32> + Clone + Debug;
```

Comment:
This name does not really tells what it is. If I was reading a code and I would see `FloatLike` I would never assume it means that it needs to implement `Sin` or `Cos`. Are all of these bounds required for the functions in `CircleSegment` impl? If not, only the required bounds should be left here. Not needed bounds are sometimes causing strange requirements.

Reply:
Yes, these are specifically the bounds required here (and there is also a long, but different set of bounds in `triangle.rs`.
I don't think there is a general name for it (or any other sub-group of traits in here). I can try to be more specific, like `SegmentInput` though.